### PR TITLE
173045-Remove Timespan for Keyvault

### PR DESCRIPTION
**Pull request by shirinbt14926**
- Throw Exception if No secret found in key Vault or Memory Cache
- Remove Caching Time for Key Vault and Fix UTs for the same
- Devops Side : Manufacturer Key Vault reference is changed

### CODE REVIEW

File: src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs
Comment (Nadeem14456) [inline src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs:1]: some minor nitpicks:

add new line after line 27.
add new line after line 33.

Refactored below method
public void CacheManufacturerKeys()
{
    _logger.LogInformation(EventIds.ManufacturerKeyCachingStart.ToEventId(), "Caching Of Manufacturer Keys started.");

    if(!string.IsNullOrEmpty(_manufacturerKeyVault.Value.ServiceUri))
    {
        var secretProperties = _secretClient.GetPropertiesOfSecrets();
        if(!secretProperties.Any())
        {
            throw new PermitServiceException(EventIds.ManufacturerIdNotFoundInKeyVault.ToEventId(), "No Secrets found in Manufacturer Key Vault");
        }

        foreach(var secretProperty in secretProperties)
        {
            GetSetManufacturerValue(secretProperty.Name);
        }
    }

    _logger.LogInformation(EventIds.ManufacturerKeyCachingEnd.ToEventId(), "Caching Of Manufacturer Keys End.");
}

**Review by Nadeem14456 — CHANGES_REQUESTED** [review]

File: src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs
[the remark this answers is not in the document]
Comment (shirinbt14926) [reply src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs:1]: Right, Removed

Comment (shirinbt14926) [reply src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs:1]: Refactored. Thanks

**Review by Nadeem14456 — APPROVED** [review]
LGTM.

File: src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs [hunk 1]
```diff
@@ -11,68 +9,63 @@ namespace UKHO.S100PermitService.Common.Services
...
             _secretClient = secretClient ?? throw new ArgumentNullException(nameof(secretClient));
-            CacheManufacturerKeys();
         }
 
         public void CacheManufacturerKeys()
         {
             _logger.LogInformation(EventIds.ManufacturerKeyCachingStart.ToEventId(), "Caching Of Manufacturer Keys started.");
-            if(!string.IsNullOrEmpty(_manufacturerKeyVault.Value.ServiceUri))
+
+            var secretProperties = _secretClient.GetPropertiesOfSecrets();
```
Comment (jrippington) [inline src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs:29]: Why are we doing this? The code should read on demand and cache accordingly?

**Review by jrippington — CHANGES_REQUESTED** [review]
Query as to why we are aggressively caching the keys? This could impact start-up time for a large vault. We should read on demand and cache accordingly.

File: src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs (hunk 1 above)
Comment (shirinbt14926) [reply src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs]: Removed

**Review by Nadeem14456 — APPROVED** [review]

**Review by amogh14893 — APPROVED** [review]
Reviewed DevOps changes.. LGTM

**Review by jrippington — APPROVED** [review]
Looks good, thanks!

**Pull request merged**